src/Idris/Parser.idr

```diff
@@ -1345,13 +1345,28 @@ recordDecl fname indents
 
 paramDecls : FileName -> IndentInfo -> Rule PDecl
 paramDecls fname indents
-    = do b <- bounds (do keyword "parameters"
-                         commit
-                         ps <- map concat (many $ typedArg fname indents)
-                         ds <- assert_total (nonEmptyBlock (topDecl fname))
-                         pure (ps, ds))
-         (ps, ds) <- pure b.val
-         pure (PParameters (boundToFC fname b) ps (collectDefs (concat ds)))
+    = do b1 <- bounds (keyword "parameters")
+         commit
+         args <- bounds (newParamDecls fname indents <|> oldParamDecls fname indents)
```

Comment:
You can `commit` again after `args` right? If the block does not parse, there is no
point in trying to parse it again by interpreting the parameters as old-style params.

Reply:
Ah yes good point!